### DISCUSSION TITLE
Doorbell press and motion switch

### DISCUSSION THREAD

**Discussion by ArjonStellinga** [dgreif/ring#619] [Q&A] [answered]
Hi,

I was looking for the doorbell and motion switch, and they are "hidden" as accessoires behind the camera feed, so found them. When I click the settings of the Ring doorbell (assuming the camera feed iis the doorbell "button"), it also mentions one accessoire, and when you click it, you will see the doorbell and the motion sensor, so two accessories?!?

I use most of my automation through Domoticz, and created a Domoticz dummy device for both of a doorbell press and a doorbell motion sensor.

Now I tried to add some automation through homekit, by switching on the Domoticz dummy motion device, when there is a Ring motion. Nothing happens.

And for the doorbell, I don't see any automation button option. Only an option for motion, not a doorbell press.

Can somebody explain to me, where and how I can automate this, and how I should look at this?

Thakns in advance!

With kind regards,
Arjon G. Stellinga

**Comment by dgreif** [comment]
For motion, you can verify that it is working properly by going into the accessories menu within the camera and watching the motion accessory as you trigger motion.  It should update within a few seconds after motion is triggered.  If it does, then the plugin is working properly and any issues are probably somewhere in your Domoticz setup.  If it doesn't, indicate that motion was triggered, let me know and I can help debug further.

Regarding the doorbell press, there should be a "Programmable Switch" within that accessories page as well.  Send me a screenshot of what you see on that page if you don't think it is there.  Within the programmable switch, you can set up an automation for a "Single Press", which will be triggered when your doorbell is pressed.

**Reply by ArjonStellinga** [reply]
Thanks!

I will have a look at what you wrote.

I do have those accessoires, you describe.

I will let you know the outcome, when I’m able to test it.

**Reply by ArjonStellinga** [reply]
I have tested the automation in HomeKit, and the Domoticz dummy switches, which are added to Homekit dit not react to the automation in HomeKit. When I test the automation, it does work, but real motion or a real doorbell press doesn’t do a thing.

I have added the pictures of my setup in this post.
![A95C4173-F61A-4FC5-A338-CF25627FC9F2](https://user-images.githubusercontent.com/66941832/116221082-96d82480-a74d-11eb-91ac-16d1ad0a022e.png)
![6FF9FAB5-58BC-4065-9156-8F50EF0B94CD](https://user-images.githubusercontent.com/66941832/116221095-99d31500-a74d-11eb-9906-62ef9539ddf4.png)
![717CC721-A11F-41CD-9200-CCD1C092113C](https://user-images.githubusercontent.com/66941832/116221101-9b044200-a74d-11eb-9728-d507b0bc5048.png)
![4AD37BD1-1528-49CA-9893-A445D9F2C2F6](https://user-images.githubusercontent.com/66941832/116221107-9b9cd880-a74d-11eb-8a5b-d638805f1d46.png)
![ACBB92CE-9C88-4226-8273-32E354CCF35B](https://user-images.githubusercontent.com/66941832/116221112-9c356f00-a74d-11eb-86bb-29a2b8ba2220.png)
![30ADC432-74E5-47A6-B227-0BFB95173DA0](https://user-images.githubusercontent.com/66941832/116221114-9cce0580-a74d-11eb-85bf-6899096a41dc.png)

**Comment by ArjonStellinga** [comment]
PS. Why did you mention it is probably an issue in my Domoticz setup?!? I activate a Dummy switch, which is presented in Homekit, nothing more, nothing less. There is no reaction from the motion, and to my knowledge, setting up a single press in Homekit tells the button what to do, if I press it once in the Home app (Action), and is not an automation of my doorbell press. And testing the automation works.

**Reply by dgreif** [reply]
The programmable switch is not "pressable" within HomeKit.  The single press is activated when the Ring doorbell is pressed.  Can you post your config for the ring plugin (without refresh token)?

**Reply by ArjonStellinga** [reply]
Everything is default, and I have played with these settings, so that's why they are in my config, I guess.: 
            "hideLightGroups": false,
            "hideDoorbellSwitch": false,
            "sendCameraMotionNotificationsToTv": false,
            "sendDoorbellMotionNotificationsToTv": false,
            "beamDurationSeconds": 60,
            "platform": "Ring"

**Comment by ArjonStellinga** [comment, marked as answer]
Looks like I got it to work, but worse, I didn’t change a thing! 😳😒😳
That’s something I don’t like at all. Anyway, the bell and motion switch on a Domoticz dummy switch now. 
I will try to see what changed in the coming week.
